lib/reservation_manager.rb

```diff
@@ -0,0 +1,67 @@
+require 'time'
+
+require_relative 'room'
+require_relative 'reservation'
+
+module Hotel
+  class ReservationManager
+    attr_reader :rooms, :reservations
+
+    def initialize(room_count)
+      @rooms = generate_rooms(room_count)
+      @reservations = []
+    end
+
+    # Parameters: 
+    # -type: symbol (:SINGLE, :BLOCK) that indicates type of reservation
+    # -range: DateRange object 
+    # -occupancy: Hash or String
+    # Returns: newly created Reservation object
+    def create_reservation(type, range, occupancy)
+      raise ArgumentError.new("Invalid reservation type (given #{type}, must be :SINGLE or :BLOCK)") unless 
+        type == :SINGLE || type == :BLOCK
+
+      occupancy = [{:room => @rooms.sample, :guest => occupancy}] if occupancy.class == String
```

Comment:
If you're going to check a type I'd recommend using `kind_of?` instead of `class` since that way you allow subclasses too:

```suggestion
      occupancy = [{:room => @rooms.sample, :guest => occupancy}] if occupancy.kind_of? String
```